kuronometer-core/src/main/scala/com/github/pedrovgs/kuronometer/Kuronometer.scala

```diff
@@ -0,0 +1,92 @@
+package com.github.pedrovgs.kuronometer
+
+import cats.free.Free
+import com.github.pedrovgs.kuronometer.KuronometerResults.{ConnectionError, KuronometerResult}
+import com.github.pedrovgs.kuronometer.app.KuronometerProgram
+import com.github.pedrovgs.kuronometer.free.algebra.{ReporterOps, ViewOps}
+import com.github.pedrovgs.kuronometer.free.domain.View.Message
+import com.github.pedrovgs.kuronometer.free.domain._
+import com.github.pedrovgs.kuronometer.free.interpreter.Interpreters
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.DurationFormatter.NanosecondsFormat.format
+import com.github.pedrovgs.kuronometer.free.interpreter.formatter.SummaryBuildStageExecutionFormatter
+
+class Kuronometer(implicit R: ReporterOps[KuronometerProgram], V: ViewOps[KuronometerProgram], interpreters: Interpreters) {
+
+  private val kuronometerHeader = "== Kuronometer =="
+
+  def reportBuildFinished(buildExecution: BuildExecution, config: Config): KuronometerResult[BuildExecution] = {
+    val filteredBuildExecution = filterBuildExecutionData(buildExecution, config)
+    val program = for {
+      result <- reportBuildExecution(filteredBuildExecution, config)
+      _ <- showReportResult(result, config)
+    } yield result
+    program.foldMap(interpreters.kuronometerInterpreter)
```

Comment:
Right now there are three entry points related to the Gradle Plugin API:

* [GetTodayBuildTimeTask](https://github.com/pedrovgs/Kuronometer/blob/3a5b6e7d6847b2e49357fbe3f2536c723f5debde/kuronometer/src/main/scala/com/github/pedrovgs/kuronometer/tasks/GetTodayBuildTimeSummaryTask.scala)
* [GetTotalBuildTimeTask](https://github.com/pedrovgs/Kuronometer/blob/3a5b6e7d6847b2e49357fbe3f2536c723f5debde/kuronometer/src/main/scala/com/github/pedrovgs/kuronometer/tasks/GetTotalBuildTimeSummaryTask.scala)
* [KuronometerBuildListener](https://github.com/pedrovgs/Kuronometer/blob/3a5b6e7d6847b2e49357fbe3f2536c723f5debde/kuronometer/src/main/scala/com/github/pedrovgs/kuronometer/KuronometerBuildListener.scala)

As this is not a library and our programs are not thought to be combined with any other algebras, I designed ``Kuronometer`` class (in my next commit is going to be an Object) to be used as a fire and forget method calls where the program is interpreted inside those methods. I think this simplifies the usage of the ``kuronometer-core`` module from the gradle plugin code. What do you think?